bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/provider/RuleResourceBundleImporter.java

```diff
@@ -148,9 +149,9 @@ protected void addNewProvidedObjects(List<String> newPortfolio, List<String> pre
     protected List<String> getPreviousPortfolio(Vendor vendor) {
         List<String> portfolio = providerPortfolio.get(vendor);
         if (portfolio == null) {
-            for (Vendor v : providerPortfolio.keySet()) {
-                if (v.getVendorSymbolicName().equals(vendor.getVendorSymbolicName())) {
-                    List<String> vendorPortfolio = providerPortfolio.get(v);
+            for (Entry<Vendor, List<String>> entry : providerPortfolio.entrySet()) {
+                if (entry.getKey().getVendorSymbolicName().equals(vendor.getVendorSymbolicName())) {
+                    List<String> vendorPortfolio = entry.getValue();
                     return vendorPortfolio == null ? List.of() : vendorPortfolio;
```

Comment:
```suggestion
                    return entry.getValue();
```

The map does not contain null values